src/main/java/io/nats/client/impl/NatsConsumerContext.java

```diff
@@ -68,7 +68,7 @@ public class NatsConsumerContext implements ConsumerContext, SimplifiedSubscript
             .replayPolicy(config.getReplayPolicy())
             .headersOnly(config.getHeadersOnly())
             .build();
-        subscribeSubject = originalOrderedCc.getFilterSubject();
+        subscribeSubject = Validator.validateSubject(originalOrderedCc.getFilterSubject(), false);
```

Comment:
validating in the constructor because it comes through here from a different code path